sway/ipc-server.c

```diff
@@ -595,6 +595,16 @@ void ipc_client_handle_command(struct ipc_client *client) {
 	switch (client->current_command) {
 	case IPC_COMMAND:
 	{
+		char *line = strtok(buf, "\n");
+		while (line) {
+			size_t line_length = strlen(line);
+			if (line + line_length >= buf + client->payload_length) {
```

Comment:
Is this check needed?

Reply:
Yes, otherwise it will replace the final null terminator with `;`. It could be changed from `>=` to `==` though.

Reply:
I thought the loop would break if it reached the end (i.e. strtok returns null).

Reply:
It will, but the following line `line[line_length] = ';'` should not be executed for the final token

Reply:
You are correct, sorry, I misremembered how strtok behaved on the last token.